Add quickstart method (#252)
ESMValGroup/ESMValTool_Tutorial: _episodes/00-quickstart.md

---
title: "Quickstart Guide"
teaching: 1
exercises: 10
questions:

- "What is the purpose of the quickstart guide?"
- "How do I activate and test the ESMValTool environment?"
- "How do I configure ESMValTool?"
- "How do I run a recipe?"

objectives:
- "Understand the purpose of the quickstart guide"
- "Activate and test the ESMValTool environment"
- "Configure ESMValTool"
- "Run a recipe"

keypoints:

- "The purpose of the quickstart guide is to enable the user to run the 
  ESMValTool as quickly as possible"
- "Use the `module load` command to activate the ESMValTool environment, 
  see the installation guide for more details, use `esmvaltool -- --help` to 
  test the ESMValTool environment"
- "Use `esmvaltool config get_config_user` to create the ESMValTool user 
  configuration file"
- "Use `esmvaltool run <recipe>.yml` to run a recipe"

> ---

Reviewer: I wonder whether we should add a note to say what version of ESMValTool this quick start guide is compatible with?
```suggestion
---
> ## Compatibility
>
> This quickstart guide is compatible with ESMValTool v2.6.0.
 {: .callout}
```

> ## What is the purpose of the quickstart guide?
>
> The purpose of the quickstart guide is to enable the user to run the 
> ESMValTool as quickly as possible by making the bare minimum number of 
> changes.
{: .discussion}

> ## How do I activate and test the ESMValTool environment?
>
> - For this quickstart guide, an assumption is made that ESMValTool has 

> >   already been installed. If this is not the case, see the 
> >   [Installation][lesson-installation] episode in this tutorial.

Reviewer: ```suggestion
>   already been installed at the site where ESMValTool will be run. If this is
>   not the case, see the [Installation][lesson-installation] episode in this
>   tutorial.
```

> 

> > - Activate the ESMValTool environment by following the instruction at

Reviewer: ```suggestion
> - Activate the ESMValTool environment by following the instructions at
```

---
title: "Quickstart guide"
teaching: 1
exercises: 10
questions:

- "What is the purpose of the quickstart guide?"
- "How do I activate and test the ESMValTool environment?"
- "How do I configure ESMValTool?"
- "How do I run a recipe?"

objectives:

- "Understand the purpose of the quickstart guide"
- "Activate and test the ESMValTool environment"
- "Configure ESMValTool"
- "Run a recipe"

keypoints:

- "The purpose of the quickstart guide is to enable a user of ESMValTool
  to run ESMValTool as quickly as possible"
- "Use the `module load` command to activate the ESMValTool environment 
  (see the installation guide for more details) and use `esmvaltool -- --help` to 
  test the ESMValTool environment"
- "Use `esmvaltool config get_config_user` to create the ESMValTool user 
  configuration file"
- "Use `esmvaltool run <recipe>.yml` to run a recipe"
---
> ## Compatibility
>
> This quickstart guide is compatible with ESMValTool v2.6.0.
 {: .callout}

> > ## What is the purpose of the quickstart guide?

Reviewer: ```suggestion

> ## What is the purpose of the quickstart guide?
```

>

> > The purpose of the quickstart guide is to enable a user of ESMValTool to run
> > ESMValTool as quickly as possible by making the bare minimum number of 
> > changes.

Reviewer: ```suggestion
> - The purpose of the quickstart guide is to enable a user of ESMValTool to
>   run ESMValTool as quickly as possible by making the bare minimum number of 
>   changes.
```

---

> title: "Quickstart Guide"

Reviewer 2: ```suggestion
title: "Quickstart guide"
```

teaching: 5

> exercises: 10

Reviewer 2: Will you add `exercises` to this lesson?

Reviewer 2: One suggestion for lesson development in the future: if learners want to practice the skills with the exercises, there should be a solution/answer. For example, when we asked them to run `esmvaltool --help`, we should also tell them what would be the output. More examples can be found in other episodes such as [03-configuration](https://esmvalgroup.github.io/ESMValTool_Tutorial/03-configuration/index.html). Also for more information see [Designing challenges](https://carpentries.github.io/curriculum-development/designing-challenges.html#designing-challenges-1).

questions:

- "How do I activate the ESMValTool environment?"

> - "How do I configure the ESMValTool?"

Reviewer 3: How do I configure ESMValTool?

- "How do I run a recipe?"

objectives:

> - "Activate the ESMValTool environment"

Reviewer 2: I could not find where this objective is explained in this lesson.

- "Configure the ESMValTool"
- "Successfully run a recipe"

keypoints:

> - "See the installation guide for activating the ESMValtool environment."

Reviewer 2: Why we point the user to the installation guide? The main idea in issue #245 is that esmvaltool is already installed.

> - "The config file has to be set up, uncommenting the environment you use."

Reviewer 2: ```suggestion
- "The information needed by ESMValTool has to be set up using `config-user.yml file`, uncommenting the Site-specific entries."
```

Reviewer 2: Please use `environment` only for `conda environment`. You can use `site` or `infrastructure` for the machine where esmvaltool is installed.

- "The command for running a recipe is `esmvaltool run`."
---

> 

Reviewer 2: Plaese add an introductory paragraph about what the learner will learn in this lesson, and who are the target audience.

> ## Installing and getting started with the ESMValTool
>
> To install the ESMValTool please see 
> [here](https://docs.esmvaltool.org/en/latest/quickstart/installation.html#)

> {: .callout}

Reviewer 2: The main idea in issue https://github.com/ESMValGroup/ESMValTool_Tutorial/issues/245 is that esmvaltool is already installed. Can we replace this with a note something like "we assume that you are working on a compute clusters where esmvaltool is installed and you have access to data. with a link to Setup https://esmvalgroup.github.io/ESMValTool_Tutorial/setup.html

> ## Configuring ESMValtool
>
> With ESMValTool installed, we can now set up our configuration file (which 
> tells ESMValTool how to run, where to get data from, and where to output it).
>
> You can find out more about this in the 

> > [documention](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/quickstart/configure.html).

Reviewer 2: Please link to the `configuration` episode instead, see https://esmvalgroup.github.io/ESMValTool_Tutorial/03-configuration/index.html

>
>
> - Ensure you are in the directory you want to get started in:
>
>   ~~~bash
>    esmvaltool config get_config_user
>    ~~~
> - Open the config-user.yml file in a text editor:
>
>   ~~~bash
>   ~/.esmvaltool/config-user.yml 
>   ~~~
> - To set up the config-user.yml file, uncomment the lines relating to the
> host you are using to run the ESMValTool. For example if you were using 
> Jasmin you should uncomment:
>
>   ```yaml
>   # Site-specific entries: Jasmin
>   # Uncomment the lines below to locate data on JASMIN
>   auxiliary_data_dir: /gws/nopw/j04/esmeval/aux_data/AUX
>   rootpath:
>     CMIP6: /badc/cmip6/data/CMIP6
>     CMIP5: /badc/cmip5/data/cmip5/output1
>     CMIP3: /badc/cmip3_drs/data/cmip3/output
>     OBS: /gws/nopw/j04/esmeval/obsdata-v2
>     OBS6: /gws/nopw/j04/esmeval/obsdata-v2
>     obs4MIPs: /gws/nopw/j04/esmeval/obsdata-v2
>     ana4mips: /gws/nopw/j04/esmeval/obsdata-v2
>     CORDEX: /badc/cordex/data/CORDEX/output
>   drs:
>     CMIP6: BADC
>     CMIP5: BADC
>     CMIP3: BADC
>     CORDEX: BADC
>     OBS: default
>     OBS6: default
>     obs4MIPs: default
>     ana4mips: default
>   ```
>

> > - Make sure to find offline and change it to true as well as setting max_parallel_tasks to 1.

Reviewer 2: Please explain why users should set these things.

{: .challenge}

> ## Running a Recipe
> 
> - To run a recipe:
> 
>   ~~~bash
>   esmvaltool run examples/recipe_python.yml 
>   ~~~
> - If everything has been set up correctly, the last line of code should look
>   something like this: 
>
>    ```yaml
>    YYYY-MM-DD HH:mm:SS, NNN UTC [93691] INFO    Run was successful
>    ```
>
> - To view the file used in this example:
>
>   ~~~bash
>    esmvaltool recipes get examples/recipe_python.yml 
>   ~~~
> 
> - Then open `recipe_python.yml` in a text editor.
>
> - You can find out more about this recipe in the 
>   [documentation](https://docs.esmvaltool.org/en/latest/recipes/recipe_examples
.html).
{: .challenge}

> > ## Outputs of a Recipie

Reviewer 2: ```suggestion
> ## Outputs of a Recipe
```

>

> > The outputs of a recipe can be found in the output_dir specified in the 

Reviewer 2: ```suggestion
> The outputs of a recipe can be found in the `output_dir` specified in the 
```
